eclass/dotnet-utils.eclass

```diff
@@ -0,0 +1,216 @@
+#
+# Example:
+# @CODE
+# NUGETS="
+# 	ImGui.NET-1.87.2
+# 	Config.Net-4.19.0
+# "
+#
+# inherit dotnet-utils
+#
+# ...
+#
+# SRC_URI="$(nuget_uris)"
+# @CODE
+
+# @FUNCTION: nuget_uris
+# @USAGE: <nuget...>
+# @DESCRIPTION:
+# Generates the URIs to put in SRC_URI to help fetch dependencies.
+# If no arguments provided, uses NUGETS variable.
+nuget_uris() {
+	local -r regex='^([a-zA-Z0-9_.-]+)-([0-9]+\.[0-9]+\.[0-9]+.*)$'
+	local nuget nugets
+
+	if (( $# != 0 )); then
+		nugets="${@}"
+	elif [[ ${NUGETS} ]]; then
+		nugets="${NUGETS}"
+	else
+		eerror "NUGETS variable is not defined and nothing passed as argument"
+		die "${FUNCNAME}: Can't generate SRC_URI from empty input"
+	fi
+
+	for nuget in ${nugets}; do
+		local name version url
+		[[ ${nuget} =~ ${regex} ]] || die "Could not parse name and version from nuget: $nuget"
+		name="${BASH_REMATCH[1]}"
+		version="${BASH_REMATCH[2]}"
+		url="https://api.nuget.org/v3-flatcontainer/${name}/${version}/${name}.${version}.nupkg"
+		echo "${url}"
+	done
+}
+
+# @FUNCTION: dotnet-utils_pkg_setup
+# @DESCRIPTION:
+# Sets up DOTNET_EXECUTABLE variables for later use in edotnet.
+dotnet-utils_pkg_setup() {
+	if use amd64; then
+		DOTNET_RUNTIME="linux-x64"
+	elif use arm; then
+		DOTNET_RUNTIME="linux-arm"
+	elif use arm64; then
+		DOTNET_RUNTIME="linux-arm64"
+	else
+		die "Unsupported arch: ${ARCH}"
+	fi
+
+	local _dotnet
+	for _dotnet in dotnet{,-bin}-${DOTNET_SLOT}; do
+		if type ${_dotnet} 1> /dev/null 2>&1; then
+			DOTNET_EXECUTABLE=${_dotnet}
+			break
+		fi
+	done
+
+}
+
+# @FUNCTION: edotnet
+# @USAGE: <command> [args...]
+# @DESCRIPTION:
+# Call dotnet, passing the supplied arguments.
+edotnet() {
+	debug-print-function ${FUNCNAME} "${@}"
+
+	local dotnet_args=(
+		-maxcpucount:$(makeopts_jobs)
+	)
+
+	edo "${DOTNET_EXECUTABLE}" "${@}" "${dotnet_args[@]}"
+}
+
+# @FUNCTION: dotnet-utils_src_unpack
+# @DESCRIPTION:
+# Unpacks the package
+dotnet-utils_src_unpack() {
+	debug-print-function ${FUNCNAME} "${@}"
+
+	local archive
+	for archive in ${A}; do
+		case "${archive}" in
+			*.nupkg)
+				;;
+			*)
+				unpack ${archive}
+				;;
+		esac
+	done
+}
+
+# @FUNCTION: dotnet-utils_src_prepare
+# @DESCRIPTION:
+# Restore the packages using 'dotnet restore'
+dotnet-utils_src_prepare() {
+	debug-print-function ${FUNCNAME} "${@}"
+
+	local restore_args=(
+		--runtime "${DOTNET_RUNTIME}"
+		--source "${DISTDIR}"
+	)
+
+	edotnet restore "${restore_args[@]}" || die "'dotnet restore' failed"
+	default_src_prepare
+}
+
+# @FUNCTION: dotnet-utils_src_compile
+# @DESCRIPTION:
+# Build the package using dotnet build
+dotnet-utils_src_compile() {
+	debug-print-function ${FUNCNAME} "${@}"
+
+	local build_args=(
+		--configuration ${DOTNET_CONFIGURATION:-$(usex debug Debug Release)}
+		--no-restore
+		--no-self-contained
+		--nologo
+		--output ${DOTNET_OUTPUT:="${WORKDIR}"/${P}_net${DOTNET_SLOT}_${DOTNET_CONFIGURATION}/${PN}}
+		-p:StripSymbols=true
+		-p:NativeDebugSymbols=false
+		-p:Version=${PV}
```

Comment:
Could you explain what `p:Version` does?

Reply:
It overwrites the version of the package, got it from an example on the dotnet documentation, and realizing now that this is most likely not necessary, removed it.